src/honeybadger/event.cr

```diff
@@ -0,0 +1,73 @@
+require "json"
+require "uri"
+
+module Honeybadger
+  struct Event
+    include JSON::Serializable
+    include JSON::Serializable::Unmapped
+
+    @[JSON::Field(key: "ts", converter: Honeybadger::Event::RFC3339Converter)]
+    getter timestamp : Time
+
+    def initialize(@timestamp = Time.utc, **properties)
+      properties.each do |key, value|
+        self[key.to_s] = value
+      end
+    end
```

Comment:
When I put this in place in my app I immediately ran into the need to build an event out of a dynamic hash instead of a named tuple:

```crystal
      hash_params = event_parameters
      event = Honeybadger::Event.new
      event["context"] = self.class.name
      event["took"] = duration.total_seconds

      hash_params.each do |key, value|
        event[key] = value
      end

      Honeybadger.send event
```

Reply:
Added support for passing the hash in https://github.com/honeybadger-io/honeybadger-crystal/pull/37/commits/6fab042d02fc4bd08d5ffbfaafd5cb17adcb16da